### PR TITLE
Quickcast Micro v1

### PR DESCRIPTION
**Pull request by Aevyz**
Works, however needs a toggle

### CODE REVIEW

**Comment by Aevyz** [comment]
Hopefully solution version 1

https://github.com/loopuleasa/Dota2-TheCore-Config-Engine/issues/152#issuecomment-211309868

File: Dota 2 Reborn Keyboard Setups/Loopuleasa's Super Compact QWERTG-DFXCV layout (ALT,SPACE mods)/dota2_gameplay_mode/dota2_keybinds_default.cfg [hunk 1]
```diff
@@ -63,17 +63,21 @@ bind "Z" "dota_recent_event;"
...
-bind "0" "+dota_control_group 9"
+bind "2" "+two"
+bind "3" "+three"
+bind "4" "+four"
+bind "5" "+five"
+bind "6" "+six"
+bind "7" "+seven"
+bind "8" "+eight"
+bind "9" "+nine"
+bind "0" "+ten"
```
Comment (FlorianWendelborn) [inline Dota 2 Reborn Keyboard Setups/Loopuleasa's Super Compact QWERTG-DFXCV layout (ALT,SPACE mods)/dota2_gameplay_mode/dota2_keybinds_default.cfg:74]: I'd suggest using a different name for that. Like `quick_control_group_10` or something smiliar, more descriptive.

Comment (Aevyz) [inline Dota 2 Reborn Keyboard Setups/Loopuleasa's Super Compact QWERTG-DFXCV layout (ALT,SPACE mods)/dota2_gameplay_mode/dota2_keybinds_default.cfg:74]: #6

**Comment by FlorianWendelborn** [comment]
Manta's version of this is nearly finished, already works in the engine.

I implemented some more variations:
- normal select (default dota)
- quick select (your version)
- view select (default dota + dota_camera_center)
- quick select + view hero (dota_camera_center onKeyUp)
- quick select + view unit (dota_camera_center onKeyDown)
- quick select + view both (dota_camera_center on both)

**Comment by Aevyz** [comment]
Interesting I'll take a look. Using manta is anyways easier than tying to edit 11 text documents and hoping you don't break anything. Do you think you could add a toggle between all the micro modes?

**Comment by FlorianWendelborn** [comment]
A ingame live-toggle? That'd be quite hard to implement and even to control. Well - if it's possible in a reasonable code-size by hand it should be possible to write a generator.

**Comment by FlorianWendelborn** [comment]
You know what - after thinking about it, I don't think it will be that hard to do. I basically just need a cycle, that changes a bound alias.

Should be able to re-use some code for that - however, making this easy to do for end-users will probably still be hard.

Only bad thing would be that it won't be easy to change it for ALL controlgroups at once, since I'd then have to allow users to change it for some too and so on...

**Pull request merged**